### PR TITLE
Adds a dialog for entering user details

### PR DESCRIPTION
**Pull request by g33kyaditya**
Built and Tested. Works as expected.

Notes:
Refinements to the QML view (for entering user details) needed

### CODE REVIEW

File: src/credentialshandler.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#include <QtCore/QMap>
+
+CredentialsHandler::CredentialsHandler(QObject* parent): QObject(parent)
+{
+
+}
+
+void CredentialsHandler::showUserCredentialsPage()
+{
+    m_view.setSource(QUrl::fromLocalFile("UserCredentials.qml"));
```
Comment (g33kyaditya) [inline src/credentialshandler.cpp]: Here, the QUrl::fromLocalFile does not work with this url however. But when I enter my "system-specific" url it works as expected. (and that is how I tested everything). This needs to be found a way around. I'm looking at Qt Resource (qrc) that might allow me to include a qml file in the executable binary itself.

Comment (Ashish-Bansal) [inline src/credentialshandler.cpp]: Don't do this stuff on backend side. This should be handled by plasmoid itself.

File: qml/datahandler.cpp [hunk 2]
```diff
@@ -19,35 +19,28 @@
...
+                                            0,
+                                            KWallet::Wallet::Synchronous
+                                          );
+    if (!m_wallet->hasFolder("KDENow")) {
+        emit showUserCredentialsPage();
     }
     else {
-        qDebug() << "Did not receive a valid reply from org.kde.kdenow /KDENow";
-        return;
+        emit addedToWallet();
```
Comment (Ashish-Bansal) [inline qml/datahandler.cpp]: Although logically code is correct but emit signal name doesn't comply with it (Nothing has been added to wallet).

File: src/daemon.cpp [hunk 3]
```diff
@@ -19,239 +19,25 @@
...
-        qDebug() << "New email has arrived, so fetch it";
-        generalGroup.writeEntry("MESSAGE_COUNT", QString::number(messageCount));
-        generalGroup.config()->sync();
-        job->stop();
-        m_session->close();
-        emit signalUpdateProcess();
+void Daemon::setCredentials(QList < UserCredentials >& credentialsList) {
+    Processor* processor = SingletonFactory::instanceFor< Processor >();
+    foreach (UserCredentials credentials, credentialsList) {
+        EmailManager manager;
```
Comment (Ashish-Bansal) [inline src/daemon.cpp]: This instance would be destroyed after returning from this function. Create a QList for EmailManager as a member variable and append dynamically allocated EmailManager to that list.

**Pull request merged**